identity/identity.go

```diff
@@ -279,7 +286,7 @@ func NewIdentity(traitsSchemaID string) *Identity {
 
 	stateChangedAt := sqlxx.NullTime(time.Now().UTC())
 	return &Identity{
-		ID:                  uuid.Nil,
+		ID:                  x.NewUUID(), // fandom
```

Comment:
I'm wondering if we should be able to use post-persist hooks. We could leave remove this change with additional pre-persist hook to validate anything on our side

Reply:
It's possible, but it would be a major change, and new kinds of data inconsistencies might occur when things fail.

It would be option 5 for https://fandom.atlassian.net/wiki/spaces/PLAT/pages/2845442110/User+account+data+inconsistencies